module/x/gravity/keeper/msg_server.go

```diff
@@ -460,3 +454,20 @@ func (k msgServer) CancelSendToEth(c context.Context, msg *types.MsgCancelSendTo
 
 	return &types.MsgCancelSendToEthResponse{}, nil
 }
+
+func (k msgServer) SubmitBadSignatureEvidence(c context.Context, msg *types.MsgSubmitBadSignatureEvidence) (*types.MsgSubmitBadSignatureEvidenceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	err := k.CheckBadSignatureEvidence(ctx, msg)
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, msg.Type()),
+			sdk.NewAttribute(types.AttributeKeyBadEthSignature, fmt.Sprint(msg.Signature)),
+			sdk.NewAttribute(types.AttributeKeyBadEthSignatureSubject, fmt.Sprint(msg.Subject)),
```

Comment:
why do you need to emit an event with the entire data? also, not sure if you want to print this in a byte format like is currently doing